.github/workflows/smoke-checks.yml

```diff
@@ -112,6 +112,8 @@ jobs:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         GITHUB_PR_NUM: ${{ github.event.number }}
         MATRIX_SIZE: ${{ strategy.job-total }}
+        IOS_SIMULATOR_DEVICE: "iPhone 14 Pro (16.4)" # TODO: delete this line as soon as Xcode 15 is stable on CI
```

Comment:
All of these TODOs are for E2E tests, and they are just spread across files. I'd like to get this one merged asap and start working on top of it when iOS/Xcode/macOS is updated on CI because it has a lot of context behind it that I would not like to keep in mind. So I'd better live with these TODOs for a couple of weeks rather than keep these PRs open.

Reply:
By the way, most of TODOs I added relate to `develop` branch and alignment with the main repo. It's just an easy way for me to keep track of these differences that should be fixed in the future when we align the flows.

Reply:
sure, if it makes the job easier for you. Two important things:
- you can easily come back to this and clean it up when ready (if comments help with that, then it's fine)
- the existing stuff works as normal